contracts/strategies/88mph/base/EightyEightMphStrategy.sol

```diff
@@ -0,0 +1,676 @@
+        uint256 depositAmount = virtualTokenTotalSupply.div(interestRate + 10**18);
+        uint256 interestAmount = virtualTokenTotalSupply.sub(depositAmount);
+
+        // 2. rollover the deposit
+        uint64 maturationTimestamp = uint64(block.timestamp + maturationTarget());
+        uint256 newDepositId;
+        (newDepositId,) = IDInterest(rewardPool()).rolloverDeposit(depositId, maturationTimestamp);
+        depositId = uint64(newDepositId);
+        require(depositId > 0, "depositId not set after rollover");
+
+        // ensure all underlying is invested now that the new deposit is available 
+        // otherwise we would take an incorrect fee for the fixed yield earnings
+        investAllUnderlying();
+
+        // 3. withdraw the interestAmount from the deposit
+        IDInterest(rewardPool()).withdraw(depositId, interestAmount, true);
+
+        uint256 rewardBalanceBefore = IERC20(rewardToken()).balanceOf(address(this));
+        // 4. convert underlying principal to reward token
+        _underlyingToRewardToken();
+
+        // 5. take fixed yield interest gains as profit
+        uint256 rewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+        uint256 feeAmount = rewardBalance.sub(rewardBalanceBefore);
+        if (feeAmount <= 0) {
+            return;
+        }
+
+        emit ProfitLogInReward(rewardBalance, feeAmount, block.timestamp);
+        IERC20(rewardToken()).safeApprove(controller(), 0);
+        IERC20(rewardToken()).safeApprove(controller(), feeAmount);
+
+        IController(controller()).notifyFee(
+            rewardToken(),
+            feeAmount
+        );
+    }
+
+    function createNewDeposit() internal {
+        // create a new deposit to get a depositId.
+        uint64 maturationTimestamp = uint64(block.timestamp + maturationTarget());
+        uint256 underlyingBalance = IERC20(underlying()).balanceOf(address(this));
+        IERC20(underlying()).approve(rewardPool(), 0);
+        IERC20(underlying()).approve(rewardPool(), underlyingBalance);
+
+        (depositId, ) = IDInterest(rewardPool()).deposit(underlyingBalance, maturationTimestamp, 0, "");
+        // ensure depositId is valid
+        require(depositId > 0, "depositId not set after deposit");
+    }
+
+    function _liquidateReward() internal {
+        // Profits can be disabled for possible simplified and rapid exit
+        if (!sell()) {
+            emit ProfitsNotCollected(sell(), true);
+            return;
+        }
+
+        _rewardsToStaked();
+        _rewardsToRewardToken();
+        _rewardsShareProfit();
+        _rewardsToUnderlying();
+    }
+
+    function _rewardsToStaked() internal {
+        // get rewards balance
+        uint256 mphBalance = IERC20(mph).balanceOf(address(this));
+        if(mphBalance <= 0) {
+            return;
+        }
+        // formula to get the correct amount for staking takes into account that the profit sharing fee would have to be
```

Comment:
we first stake the rewards and then take the profit sharing fee. 
Amounts are corrected through some basic math to ensure profitSharingFee is deducted from the initial rewards amount